mithril-aggregator/src/runtime/state_machine.rs

```diff
@@ -281,22 +307,26 @@ impl AggregatorRuntime {
                 nested_error: None,
             })?;
 
-        // Every error raised below this comment must be raised to critical in order to trigger an aggregator restart which will avoid a dead loop in the state machine
-        // TODO: Implement a graceful retry mechanism for artifact creation and label open messages with `is_artifact_created` (to keep track of certified but without artifacts open messages)
-        self.runner.drop_pending_certificate().await.map_err(|e| {
-            RuntimeError::critical(
-                "transiting SIGNING → READY: failed dropping pending certificate",
-                Some(e),
-            )
-        })?;
+        // Every error raised below this comment must be raised to critical in
+        // order to trigger an aggregator restart which will avoid a dead loop
+        // in the state machine TODO: Implement a graceful retry mechanism for
+        // artifact creation and label open messages with `is_artifact_created`
+        // (to keep track of certified but without artifacts open messages)
+        self.runner
+            .drop_pending_certificate()
+            .await
+            .map_err(|e| RuntimeError::ReInit {
+                message: "transiting SIGNING → READY: failed to drop pending certificate"
+                    .to_string(),
+                nested_error: Some(e),
+            })?;
         self.runner
             .create_artifact(&state.open_message.signed_entity_type, &certificate)
             .await
-            .map_err(|e| {
-                RuntimeError::critical(
-                    "transiting SIGNING → READY: failed creating artifact",
-                    Some(e),
-                )
+            .map_err(|e| RuntimeError::ReInit {
+                message: "transiting SIGNING → READY: failed to create artifact. Retrying…"
```

Comment:
The retry likely occurred in the `create_artifact` function. Now the state machine will be re-initialized and there won't be any more retry on this artifact:
```suggestion
                message: "transiting SIGNING → READY: failed to create artifact"
```